src/components/chatX/ChatXPage.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { Link } from 'react-router';
+import checkAuth from '../requireAuth';
+import ChatXBody from './ChatXBody';
+import ChatXInput from './ChatXInput';
+import { messageSent, getInitial10Messages, getNewMessages, getMessageListStartAtKey } from '../../actions/chatXActions';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+export class ChatXPage extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.user = this.props.user.email;
+
+        this.state = {
+            messageList: [],
+            newMessage: ""
+        };
+
+        this.newMsgCallback = this.newMsgCallback.bind(this);
+        this.getInitialMessageList = this.getInitialMessageList.bind(this);
+        this.updateInputState = this.updateInputState.bind(this);
+        this.sendMessage = this.sendMessage.bind(this);
+        this.setScrollToBottom = this.setScrollToBottom.bind(this);
+    }
+
+    componentDidMount() {
+        this.props.actions.getNewMessages(this.newMsgCallback);
+        this.getInitialMessageList();
+    }
+
+    newMsgCallback(msg) {
+        this.props.actions.getMessageListStartAtKey(this.props.chatX.firstKey)
+            .then(() => {
+                this.setState({ messageList: this.props.chatX.msgList });
+                this.setScrollToBottom();
```

Comment:
If you want to follow the React "way-of-life" (lol), that's not exactly how you'd do it.

You'd rather hook into the [`componentDidUpdate`](https://reactjs.org/docs/react-component.html#componentdidupdate), compare previous and current state to identify if a message was added and scroll to it if it's the case.